lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2108,29 +2112,31 @@ public function createEntity($className, array $data, &$hints = [])
                 isset($hints[Query::HINT_REFRESH])
                 && isset($hints[Query::HINT_REFRESH_ENTITY])
                 && ($unmanagedProxy = $hints[Query::HINT_REFRESH_ENTITY]) !== $entity
-                && $unmanagedProxy instanceof Proxy
+                && $unmanagedProxy instanceof GhostObjectInterface
                 && $this->isIdentifierEquals($unmanagedProxy, $entity)
             ) {
+                // @TODO messy, but for now we try continuing operation as normal
+                // @TODO ideally, all this scenario should be completely gone
+                $entity = $unmanagedProxy;
                 // DDC-1238 - we have a managed instance, but it isn't the provided one.
                 // Therefore we clear its identifier. Also, we must re-fetch metadata since the
                 // refreshed object may be anything
-                foreach ($class->identifier as $fieldName) {
-                    $property = $class->getProperty($fieldName);
+                // @TODO DDC-1238 seems to be invalid.
```

Comment:
This refers to the [`DDC1238Test`](https://github.com/doctrine/doctrine2/blob/1d5c87ee4bcf22d699530b3792739f33b91cf6cd/tests/Doctrine/Tests/ORM/Functional/Ticket/DDC1238Test.php)